src/datachain/data_storage/sqlite.py

```diff
@@ -333,16 +333,30 @@ def create_table(
         if_not_exists: bool = True,
         *,
         kind: str | None = None,
-    ) -> None:
+    ) -> bool:
+        """Create table and return True if created, False if already existed."""
+        table_existed = self.has_table(table.name)
         self.execute(CreateTable(table, if_not_exists=if_not_exists))
+        return not table_existed
 
     def drop_table(self, table: "Table", if_exists: bool = False) -> None:
         self.execute(DropTable(table, if_exists=if_exists))
+        # Remove the table from metadata to avoid stale references
+        if table.name in self.metadata.tables:
+            self.metadata.remove(table)
 
     def rename_table(self, old_name: str, new_name: str):
         comp_old_name = quote_schema(old_name)
         comp_new_name = quote_schema(new_name)
-        self.execute_str(f"ALTER TABLE {comp_old_name} RENAME TO {comp_new_name}")
+        try:
+            self.execute_str(f"ALTER TABLE {comp_old_name} RENAME TO {comp_new_name}")
+        except Exception as e:
+            raise RuntimeError(
+                f"Failed to rename table from '{old_name}' to '{new_name}': {e}"
+            ) from e
+        # Remove old table from metadata to avoid stale references
+        if old_name in self.metadata.tables:
```

Comment:
where do we need this? do we use this from multiple threads somewhere? (here is is possible race condition with this implementation , it is not thread safe)